File: src/algorithms/check-record.js
/**
 * Given a string s representing an attendance record for a student where each
 * character signifies whether the student was absent, late, or present on that
 * day. The record contains the following three characters:
 *   * 'A': absent, 'L': late, 'P': present.
 * The student is eligible for an attendance award if they were absent for fewer
 * than 2 days, and the studen was never late for 3 or more consecutive days.
 * 
 * Returns true if the student is eligible for an attendance award.
 * 
 * @param {string} s - A string made up of the characters 'A', 'L', and 'P'
 * @returns {boolean} - Return true if the student is eligible for an award
 */
function checkRecord (s) {
  let daysLate = 0;
  let daysAbsent = 0;
  for (let i = 0; i < s.length; i++) {
    if (s[i] === 'A') {
      daysAbsent++;
      daysLate = 0;
      if (daysAbsent > 1) {
        return false;
      }
    } else if (s[i] === 'L') {
      daysLate++;
      if (daysLate >= 3) {
        return false;
      }
    } else {
      daysLate = 0;
    }
  }
  return true;
}

File: src/algorithms/check-record.py
def checkRecord(s):
    """You are given a string s representing an attendance record for a student 
    where each character signifies whether the student was absent, late, or 
    present on that day. The record only contains the following three 
    characters:
    • 'A': Absent.
    • 'L': Late.
    • 'P': Present.
    The student is eligible for an attendance award if they meet both of the 
    following criteria:
    • The student was absent ('A') for strictly fewer than 2 days total.
    • The student was never late ('L') for 3 or more consecutive days.
    Return true if the student is eligible for an attendance award, or false 
    otherwise.

    Args:
        s (str): A string of "A", "L", and "P" characters.

    Returns:
        Boolean: Returns true if a student is eligible for an attendance award.
    """
    absences = 0
    late = 0
    for char in s:
        if char == "A":
            absences += 1
            late = 0
        elif char == "L":
            late += 1
        else:
            late = 0
        if absences > 1 or late > 2:
            return False
    return True
